lib/miq_expression.rb

```diff
@@ -566,8 +566,8 @@ def self._to_ruby(exp, context_type, tz)
 
   def to_sql(tz = nil)
     tz ||= "UTC"
-    @pexp, attrs = preprocess_for_sql(@exp.deep_clone)
-    sql = to_arel(@pexp, tz).to_sql if @pexp.present?
+    sqlable_expression_data, attrs = preprocess_for_sql(@exp.deep_clone)
+    sql = Component.build(sqlable_expression_data).to_sql(tz) if sqlable_expression_data.present?
```

Comment:
Aye. It's in this method for now because to_sql doesn't support all the operators, and requires the preprocessing in the line above. That method should also be able to go away